### PR TITLE
Adding bounds checks and key checks for appcompat shims table

### PR DESCRIPTION
**Pull request by poppyseedplehzr**
This adds checks for keys as well as bounds checks for substr usage. I'll add a test plan after creating some garbage registry values and ensuring this table handles them appropriately.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 23caf9185b6ca5a318bc0d004df52e5390a99466 (Job results: [1359](https://jenkins.osquery.io/job/osqueryPullRequestAudit/1359/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 334af665f1afc1b46fe5299c0f5328b167103628 (Job results: [660](https://jenkins.osquery.io/job/osqueryPullRequestOSX/660/)) failed one or more tests (**macOS/OS X**).

File: osquery/tables/system/windows/appcompat_shims.cpp [hunk 1]
```diff
@@ -107,5 +118,5 @@ QueryData genShims(QueryContext& context) {
 
   return results;
 }
-}
-}
+} // namespace tables
```
Comment (poppyseedplehzr) [inline osquery/tables/system/windows/appcompat_shims.cpp]: Are we doing this now? Or is this just my `clang-format` freakin out?

**Comment by poppyseedplehzr** [comment]
Tested this with some garbage registry data, looks to be working fine now.

**Comment by poppyseedplehzr** [comment]
I followed the walkthrough [here](https://www.blackhat.com/docs/eu-15/materials/eu-15-Pierce-Defending-Against-Malicious-Application-Compatibility-Shims-wp.pdf) to install an appcompat shim, and then verified that we're querying it correctly.
```
osquery> select * from appcompat_shims;
+----------------+--------------------------------------------------------------------------------+-------------+--------------+-------+----------------------------------------+
| executable     | path                                                                           | description | install_time | type  | sdb_id                                 |
+----------------+--------------------------------------------------------------------------------+-------------+--------------+-------+----------------------------------------+
| Clockres64.exe | C:\WINDOWS\AppPatch\Custom\Custom64\{f9de3e42-e1cc-4669-aecf-140cc22e3bff}.sdb | dataset1    | 1492372807   | 65536 | {f9de3e42-e1cc-4669-aecf-140cc22e3bff} |
+----------------+--------------------------------------------------------------------------------+-------------+--------------+-------+----------------------------------------+
```

**Pull request merged**